src/mscorlib/shared/System/Version.cs

```diff
@@ -333,13 +333,13 @@ private static Version ParseVersion(ReadOnlySpan<char> input, bool throwOnFailur
             // Find the ends of the optional minor and build portions.
             // We musn't have any separators after build.
             int buildEnd = -1;
-            int minorEnd = input.IndexOf('.', majorEnd + 1);
+            int minorEnd = input.Slice(majorEnd + 1).IndexOf('.');
```

Comment:
As with the other cases where slicing was added, I don't think this and the below cases do the right thing.